### PR TITLE
ENH: Increase coverage for `itk::FancyString`

### PR DESCRIPTION
**Pull request by jhlegarreta**
- STYLE: Rename test to honor the tested class
- ENH: Increase coverage for `itk::FancyString`
- STYLE: Use testing macros in `itkFancyStringTest.cxx`

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Not sure about the segfault:
https://open.cdash.org/viewTest.php?onlyfailed&buildid=9125424

File: Modules/IO/XML/test/itkFancyStringTest.cxx [hunk 1]
```diff
@@ -38,6 +38,13 @@ testFancyStringWithItkArray();
 void
 testFancyStringForStringOperations();
 
+int
```
Comment (dzenanz) [inline Modules/IO/XML/test/itkFancyStringTest.cxx:42]: In commit message, `FrancyString` -> `FancyString`

File: Modules/IO/XML/test/itkFancyStringTest.cxx [hunk 2]
```diff
@@ -443,3 +454,65 @@ testFancyStringForStringOperations()
...
+{
+  {
+    itk::FancyString  s1{ "Hello World!" };
+    const std::string s2{ "Hello World" };
+
+    ITK_TEST_EXPECT_TRUE(s1 != s2);
+
+    const std::string s3{ "Hello world!" };
+
+    ITK_TEST_EXPECT_TRUE(s1 != s3);
```
Comment (dzenanz) [inline Modules/IO/XML/test/itkFancyStringTest.cxx:452]: How about adding `ITK_TEST_EXPECT_FALSE(s1 != "Hello World!");`

File: Modules/IO/XML/test/itkFancyStringTest.cxx [hunk 3]
```diff
@@ -443,3 +454,65 @@ testFancyStringForStringOperations()
...
+}
+
+int
+testFancyStringEqualOperator()
+{
+  {
+    itk::FancyString  s1{ "Hello World!" };
+    const std::string s2{ "Hello World!" };
+
+    ITK_TEST_EXPECT_TRUE(s1 == s2);
```
Comment (dzenanz) [inline Modules/IO/XML/test/itkFancyStringTest.cxx:483]: Add `ITK_TEST_EXPECT_FALSE(s1 == "Hello!");`

File: Modules/IO/XML/test/itkFancyStringTest.cxx (hunk 1 above)
Comment (jhlegarreta) [reply Modules/IO/XML/test/itkFancyStringTest.cxx:42]: :+1: Done in 40b091c.

File: Modules/IO/XML/test/itkFancyStringTest.cxx (hunk 2 above)
Comment (jhlegarreta) [reply Modules/IO/XML/test/itkFancyStringTest.cxx:454]: Done in 40b091c.

File: Modules/IO/XML/test/itkFancyStringTest.cxx (hunk 3 above)
Comment (jhlegarreta) [reply Modules/IO/XML/test/itkFancyStringTest.cxx:487]: Done in 40b091c.

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
Rebase to resolve conflicts.

**Review by thewtex — APPROVED** [review]
:100:

**Pull request merged**